BrainPortal/app/models/cluster_task.rb

```diff
@@ -1007,48 +1023,131 @@ def prerequisites_fulfilled?(for_state)
+  end
+
+  # Returns the basename of the captured standard error of
+  # the scientific script.
+  def science_stderr_basename(run_number=nil) #:nodoc:
+    "#{SCIENCE_STDERR_BASENAME}.#{self.name}.#{self.run_id(run_number)}"
+  end
+
+  # Returns the basename for the QSUB standard output capture file for the task.
+  # This is not a full path, just a filename relative to the work directory.
+  # The file itself is not garanteed to exist.
+  def qsub_stdout_basename(run_number=nil) #:nodoc:
+    "#{QSUB_STDOUT_BASENAME}.#{self.name}.#{self.run_id(run_number)}"
+  end
+
+  # Returns the basename for the QSUB standard error capture file for the task.
+  # This is not a full path, just a filename relative to the work directory.
+  # The file itself is not garanteed to exist.
+  def qsub_stderr_basename(run_number=nil) #:nodoc:
+    "#{QSUB_STDERR_BASENAME}.#{self.name}.#{self.run_id(run_number)}"
+  end
+
+  # Given two basenames for two files, will combine them into a
+  # a single report; the report will be the content of the
+  # first file  where the token +__CBRAIN_CAPTURE_PLACEHOLDER__+
+  # will be replaced by the content of the second file. The
+  # new report is saved in a file with a name set to a modified version
+  # of the name of the first file, "qsub_outerr-combined".
+  #
+  # The method returns that filename. If the combined report already exists,
+  # the method just returns the filename, it will not be recreated.
+  #
+  # If the first file doesn't exist or is not complete (the content must
+  # end with the line "CBRAIN Task Exiting" then the filename returned
+  # is just the second file in argument, +science_outerr+ ; this means
+  # that while a task is running, for instance, this method will still
+  # return the output as it is being created.
+  #
+  # For this method to work properly, the CWD must be set to the
+  # task's work directory.
+  def find_or_create_combined_file(qsub_outerr, science_outerr) #:nodoc:
+    cb_error "Current directory is not the task's work directory?" unless self.we_are_in_workdir
+
+    # Report already prepared? Just return its basename
+    combined_file = "#{qsub_outerr}-combined"
+    return combined_file  if File.exists?(combined_file)
+
+    # Task outputs incomplete? Point to science output captured
+    return nil            if ! File.exists?(science_outerr)
+    return science_outerr if ! File.exists?(qsub_outerr)
+    qsub_content = check_task_ending_keyword(qsub_outerr) # returns nil unless content contains "CBRAIN Task Exiting"
+    return science_outerr if qsub_content.blank?
+
+    # Create combined report
+    science_content = File.read(science_outerr) # can be big but generally not to much
+    qsub_content.sub!('__CBRAIN_CAPTURE_PLACEHOLDER__',science_content)
+    File.open("#{combined_file}-tmp#{Process.pid}","w") { |fh| fh.write(qsub_content) }
+    File.rename("#{combined_file}-tmp#{Process.pid}", combined_file) # atomicity for the win
+    combined_file
```

Comment:
Yes, I considered doing that, but the difference is minimal: about 1 line at he beginning, and 5-6 lines at the end. And I think seeing the the content of the qsub output in the interface can be useful for the admin, but adding separate tabs will be darn confusing for the users...

It could be done, but I'd have to add two new fields to the Controls object that shuffles information between the Bourreau and the Portal, and that's a layer I don't like to touch.